vmdb/app/views/layouts/_exp_editor.html.haml

```diff
@@ -0,0 +1,98 @@
+                    :title                 => 'Undo the previous change')
+      - else
+        %li.dimmed
+          = image_tag("/images/toolbars/undo.png")
+      - if @edit[@expkey][:exp_idx] < @edit[@expkey][:exp_array].length - 1
+      %li
+        = link_to(image_tag('/images/toolbars/redo.png',
+                            :alt => 'Redo the next change'),
+                  {:action  => 'exp_button',
+                   :pressed => 'redo'},
+                  "data-miq_sparkle_on"  => true,
+                  "data-miq_sparkle_off" => true,
+                  :remote                => true,
+                  :title                 => 'Redo the next change')
+      - else
+        %li.dimmed
+          = image_tag("/images/toolbars/redo.png")
+
+      %span#exp_buttons_off
+        - %w(and or not discard).each do |image|
+          %li.dimmed
+            = image_tag("/images/toolbars/#{image}.png")
+      %span#exp_buttons_not{:style =>"display:none"}
+        - [[_("AND with a new expression element"), 'and',     'and'],
+           [_("OR with a new expression element"),  'or',      'or'],
+           ["",                                     'not',     ''],
+           [_("Remove this expression element"),    'discard', 'remove']].each do |title image pressed|
+          %li
+            - if title.empty?
+              = link_to(image_tag("/images/toolbars/#{image}.png")
+            - else
+              = link_to(image_tag("/images/toolbars/#{image}.png"
+                                  :alt                   => title,
+                                  {:action  => 'exp_button',
+                                   :pressed => pressed},
+                                  "data-miq_sparkle_on"  => true,
+                                  "data-miq_sparkle_off" => true,
+                                  :remote                => true,
+                                  :title                 => title)
+      %span#exp_buttons_on{:style => "display:none"}
+        - [[_("AND with a new expression element"),       'and',     'and'],
+           [_("OR with a new expression element"),        'or',      'or'],
+           [_("Wrap this expression element with a NOT"), 'not',     'not'],
+           [_("Remove this expression element"),          'discard', 'remove']].each do |title image pressed|
+          %li
+            = link_to(image_tag("/images/toolbars/#{image}.png",
+                                :alt => title),
+                      {:action  => 'exp_button',
+                       :pressed => pressed},
+                      "data-miq_sparkle_on"  => true,
+                      "data-miq_sparkle_off" => true,
+                      :remote                => true,
+                      :title                 => title)
+    %div{:style => "padding:10px"}
+      - @edit[@expkey][:exp_table].each do |token| %>
+        - if ! ["AND", "OR", "(", ")", "???"].include?([token].flatten.first)
```

Comment:
uff, what's that `.include?([token].flatten.first)` parse error in my head, why is that?